stl/inc/vector

```diff
@@ -489,6 +489,8 @@ private:
 public:
     static_assert(!_ENFORCE_MATCHING_ALLOCATORS || is_same_v<_Ty, typename _Alloc::value_type>,
         _MISMATCHED_ALLOCATOR_MESSAGE("vector<T, Allocator>", "T"));
+    static_assert(!is_reference_v<_Ty>, "The C++ Standard forbids containers of reference elements "

```

Comment:
I believe we should only need the one in allocator, similar to how `const T` is handled

Reply:
I thought likewise but at the same time, I thought that even with a custom allocator the requirements persist. In the case of this allocator static assert, it is to prevent ill formation, but mine are to clarify compile-time error messages.